### DISCUSSION TITLE
why use X_train, X_test = train_test_split(X, test_size=0.33, shuffle=False)

### DISCUSSION THREAD

**Discussion by Jacques2101** [skfolio/skfolio#95] [Q&A]
Hi, i am new at using Skfolio and I would like to understand why I should use : 
`X_train, X_test = train_test_split(X, test_size=0.33, shuffle=False)`
and what is the difference between .fit and .predict ? 

I don't understand in what purpose we have to do that ? 

Thx for your help.

**Comment by HugoDelatte** [comment]
Hello,

The `fit` and `predict` methods are widely-used API conventions across many libraries (e.g., Scikit-Learn, TensorFlow, Keras, XGBoost, etc.).

- `fit` **trains** your model on the training dataset (`X_train`). In our case, it finds the optimal asset weights.
- `predict` **makes predictions** on new data (`X_test`). Here, it uses the asset weights found during `fit` to apply them to the new price return data, allowing you to evaluate your portfolio’s performance out-of-sample.

Using this API convention lets you leverage the entire data science toolkit provided by Scikit-Learn, such as cross-validation and parameter tuning.

`X_train, X_test = train_test_split(X, test_size=0.33, shuffle=False)` splits the data into a training set and a test set. While not strictly necessary, this split can be helpful to see how your portfolio performs on new data that wasn’t used in the optimization (e.g., it can help detect overfitting).

These examples may be helpful: https://skfolio.org/auto_examples

Hope it helps!

**Reply by Jacques2101** [reply]
OK Thx.  So predict is just for backtesting the strategy that has been estimated with fit instructions

**Reply by HugoDelatte** [reply]
That's correct
